hostpool.go

```diff
@@ -38,269 +27,40 @@ type standardHostPoolResponse struct {
 // allow you to Get a HostPoolResponse (which includes a hostname to use),
 // get the list of all Hosts, and use ResetAll to reset state.
 type HostPool interface {
+	// Get selects a host from the pool. The returned response should then be marked as
+	// either successful or failed when the host is used (by calling Mark() on the response
+	// with the error value either nil for success, or set for failure).
 	Get() HostPoolResponse
-	// keep the marks separate so we can override independently
-	markSuccess(HostPoolResponse)
-	markFailed(HostPoolResponse)
 
+	// ResetAll marks all hosts in the pool as alive
 	ResetAll()
+
 	// ReturnUnhealthy when called with true will prevent an unhealthy node from
 	// being returned and will instead return a nil HostPoolResponse. If using
 	// this feature then you should check the result of Get for nil
 	ReturnUnhealthy(v bool)
+
+	// Hosts returns the current set of hosts known to the host pool
 	Hosts() []string
+
+	// SetHosts sets the current set of hosts known to the host pool. The set hosts default
+	// to being alive
 	SetHosts([]string)
 
+	// Statistics returns a sample of properties of the hostpool that would be useful
+	// to monitor, for example, the number of hosts, and the number of alive hosts
+	Statistics() HostPoolStatistics
```

Comment:
By adding this to the interface, the epsilon implementation will no longer satisfy it - I think we should implement this method on that type to avoid problems (since we allow it to be used via config).